src/mongoChangeStream.ts

```diff
@@ -94,7 +94,7 @@ export function initSync<ExtendedEvents extends EventEmitter.ValidEventTypes>(
 
   /** Detect if resync flag is set */
   const detectResync = (resyncCheckInterval = ms('1m')) => {
-    let resyncTimer: NodeJS.Timer
+    let resyncTimer: ReturnType<typeof setInterval>
```

Comment:
Is there a reason/benefit to use the ReturnType util type to infer the timer type here as opposed to just setting it to `NodeJS.Timer`

Reply:
TypeScript was complaining for some reason. In general, this is kind of nice pattern, though, since the return type for `setTimeout` and `setInterval` are slightly different depending on whether you are in a Node environment or browser environment.

Reply:
I was able to change this to `NodeJS.Timeout` which might make the most sense here considering this library isn't meant to be used in a browser environment.